### PR TITLE
NETOBSERV-1595: generate BPF from CI

### PR DESCRIPTION
**Pull request by jotak**
- Add BPF generation steps in the main Dockerfile
- Split `make generate` in 2 (one for BPF another for protoc ; we don't need to build protobuf from CI as this is still code)
- Add a pull-request check to verify that the provided BPF bin match the code

### CODE REVIEW

**Comment by jotak** [comment]
I intentionally removed the .o files from source and [generated a build](https://github.com/netobserv/netobserv-ebpf-agent/pull/309#issuecomment-2033646154) just to make sure it uses CI-generated .o
I'll put back .o in git so that they are still available for local runs.

File: Dockerfile [hunk 1]
```diff
@@ -2,14 +2,29 @@
 ARG TARGETPLATFORM=linux/amd64
 ARG BUILDPLATFORM=linux/amd64
 ARG TARGETARCH=amd64
+ARG GOVERSION="1.21.7"
 
-# Build the manager binary
-FROM --platform=$BUILDPLATFORM docker.io/library/golang:1.21 as builder
-
+# Build layer
+FROM --platform=$TARGETPLATFORM fedora:38 as builder
```
Comment (msherif1234) [inline Dockerfile:8]: Pls use fc39 as in the generator your pr should have no .o changes

File: Makefile [hunk 2]
```diff
@@ -114,20 +114,27 @@ lint: prereqs ## Lint the code
...
-.PHONY: generate
-generate: export BPF_CLANG := $(CLANG)
-generate: export BPF_CFLAGS := $(CFLAGS)
-generate: prereqs ## Generate artifacts of the code repo (pkg/ebpf and pkg/proto packages)
+.PHONY: gen-bpf
+gen-bpf: export BPF_CLANG := $(CLANG)
+gen-bpf: export BPF_CFLAGS := $(CFLAGS)
+gen-bpf: ## Generate BPF (pkg/ebpf package)
 	@echo "### Generating BPF Go bindings"
+	test -f $(shell go env GOPATH)/bin/bpf2go || go install github.com/cilium/ebpf/cmd/bpf2go@${CILIUM_EBPF_VERSION}
```
Comment (msherif1234) [inline Makefile:122]: U can do preqs here instead of repeating install

**Pull request merged**